grant-project/grant-monorepo
# Cypress e2e basics

### What this does
- adds `e2e/` directory containing cypress e2e
- implements a basic `browse` test that visits every page
- implements a `create` test -- currently this fails on "publish" as there is an issue with the injected web3, I have deferred solving this for the time being knowing that much will be changing in the near future

### What this doesn't do
- CI, probably best to leave this out for now
- MetaMask (in general cypress can't really interact with extensions properly)

### Testing
1. go to `e2e` directory
1. run `yarn`
1. make sure dev environment is running, then `yarn cypress:open`
1. run each of the tests from the cypress UI which appears

### Demo
![e2e-demo](https://user-images.githubusercontent.com/11430954/48025890-329ef180-e10a-11e8-9674-67c7fd573d13.gif)
